packages/synapse-react-client/src/components/ChallengeSubmission/ChallengeSubmissionStepper.tsx

```diff
@@ -0,0 +1,206 @@
+
+enum StepsEnum {
+  SELECT_COMMIT = 'SELECT_COMMIT',
+  SELECT_EVALUATION = 'SELECT_EVALUATION',
+  SUBMISSION_SUCCESS = 'SUBMISSION_SUCCESS',
+}
+type StepKey = keyof typeof StepsEnum
+type StepList = {
+  [key in StepKey]: Step
+}
+
+const stepsDocker: StepList = {
+  SELECT_COMMIT: {
+    id: StepsEnum.SELECT_COMMIT,
+    title: 'Select Commit for Submission',
+    nextStep: StepsEnum.SELECT_EVALUATION,
+    nextEnabled: false,
+  },
+  SELECT_EVALUATION: {
+    id: StepsEnum.SELECT_EVALUATION,
+    title: 'Select Evaluation Queue',
+    previousStep: StepsEnum.SELECT_COMMIT,
+    confirmStep: StepsEnum.SUBMISSION_SUCCESS,
+    confirmButtonText: 'Submit',
+  },
+  SUBMISSION_SUCCESS: {
+    id: StepsEnum.SUBMISSION_SUCCESS,
+    title: 'Submission Successful!',
+    confirmButtonText: 'Close',
+    confirmEnabled: true,
+  },
+}
+
+const stepsFile: StepList = {
+  SELECT_COMMIT: { id: StepsEnum.SELECT_COMMIT, title: '' },
+  SELECT_EVALUATION: {
+    id: StepsEnum.SELECT_EVALUATION,
+    title: 'Select Evaluation Queue',
+    confirmStep: StepsEnum.SUBMISSION_SUCCESS,
+    confirmButtonText: 'Submit',
+  },
+  SUBMISSION_SUCCESS: {
+    id: StepsEnum.SUBMISSION_SUCCESS,
+    title: 'Submission Successful!',
+    confirmButtonText: 'Close',
+    confirmEnabled: true,
+  },
+}
+
+const getSteps = (entityType: EntityType.DOCKER_REPO | EntityType.FILE) => {
+  return entityType === EntityType.DOCKER_REPO ? stepsDocker : stepsFile
+}
+
+type ChallengeSubmissionStepperProps = {
+  projectId: string
+  userId: string
+  teamId: string | undefined
+  entity: EntityItem
+  entityType: EntityType.DOCKER_REPO | EntityType.FILE
+  isShowingModal: boolean
+  onClose: () => void
+}
+
+function ChallengeSubmissionStepper({
+  projectId,
+  userId,
+  teamId,
+  entity,
+  entityType,
+  isShowingModal,
+  onClose,
+}: ChallengeSubmissionStepperProps) {
+  const { accessToken } = useSynapseContext()
+  const steps = getSteps(entityType)
+  const initialStep =
+    entityType === EntityType.DOCKER_REPO
+      ? steps.SELECT_COMMIT
+      : steps.SELECT_EVALUATION
+  const [step, setStep] = useState<Step>(initialStep)
+  const [errorMessage, setErrorMessage] = useState<string>()
+  const [selectedCommit, setSelectedCommit] = useState<DockerCommit>()
+
+  const [submissionName, setSubmissionName] = useState<string>('')
+  const [selectedEval, setSelectedEval] = useState<string | undefined>()
+  const [submissionError, setSubmissionError] = useState<string | undefined>()
+  const [confirming, setConfirming] = useState<boolean>(false)
+  const [loading, setLoading] = useState<boolean>(false)
+
+  const submitRepoForEvaluation = async () => {
+    if (!entity.id || !selectedCommit) return
+
+    const submission: EvaluationSubmission = {
+      userId: userId,
+      evaluationId: selectedEval!,
+      entityId: entity.id,
+      dockerRepositoryName: entity.repositoryName,
+      dockerDigest: selectedCommit.digest,
+      versionNumber: 1,
```

Comment:
As you mentioned in slack, this version number should be the version returned with the entity.
Also need to set contributors.
https://rest-docs.synapse.org/rest/org/sagebionetworks/evaluation/model/Submission.html

Reply:
Added the `entity.versionNumber` if defined, otherwise defaulting to 1 (versionNumber is required on `EvaluationSubmission`)

Reply:
It's better to fail than to submit the wrong entity version to the evaluation queue.  You could default to -1 and probably let the service call fail

Reply:
I don't actually see entity.versionNumber being used here, am I missing something?